### PR TITLE
Set build type for PyTorch explicitely

### PR DESCRIPTION
**Pull request by Flamefire**
PyTorch checks these environment variables in this order to determine the build type (for CMake) to use:
- CMAKE_BUILD_TYPE
- DEBUG
- REL_WITH_DEB_INFO

If we don't set it explicitely we might end up with an unexpected debug build.
Add the `build_type`EC option used in the CMakeMake easyblock with similar semantics.

### CODE REVIEW

File: easybuild/easyblocks/p/pytorch.py [hunk 1]
```diff
@@ -407,6 +409,23 @@ def add_enable_option(name, enabled):
...
+        if build_type is None:
+            build_type = 'Debug' if self.toolchain.options.get('debug', None) else 'Release'
+        else:
+            for name in ('prebuildopts', 'preinstallopts', 'custom_opts'):
+                if '-DCMAKE_BUILD_TYPE=' in self.cfg[name]:
+                    self.log.warning('CMAKE_BUILD_TYPE is set in %s. Ignoring build_type', name)
+                    build_type = None
+        if build_type:
+            if pytorch_version >= '1.2.0':
+                options.append('CMAKE_BUILD_TYPE', build_type)
```
Comment (SebastianAchilles) [inline easybuild/easyblocks/p/pytorch.py:422]: I am getting the following:
```
TypeError: append() takes exactly one argument (2 given)
```

```suggestion
                options.append('CMAKE_BUILD_TYPE=' + build_type)
```

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _PyTorch-1.12.0-foss-2022a-CUDA-11.7.0.eb_ 
 * **SUCCESS** _PyTorch-2.1.2-foss-2022b.eb_ 
 * **SUCCESS** _PyTorch-1.12.0-foss-2022a.eb_ 

Build succeeded for 3 out of 3 (3 easyconfigs in total)
bwd-rockylinux-94 - Linux Rocky Linux 9.4 (Blue Onyx), x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), 2 x NVIDIA NVIDIA GeForce GTX 1060 6GB, 535.161.08, Python 3.9.18
See https://gist.github.com/SebastianAchilles/6691cc73b7a496b696032d73a0e6a00d for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _PyTorch-2.1.2-foss-2023a-CUDA-12.1.1.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
skl-rockylinux-810 - Linux Rocky Linux 8.10, x86_64, Intel(R) Core(TM) i7-10700 CPU @ 2.90GHz (skylake), 1 x NVIDIA NVIDIA RTX A4000, 555.42.02, Python 3.6.8
See https://gist.github.com/SebastianAchilles/e350ef77f6ffb4691339c118ebd2a87d for a full test report.

**Review by ocaisa — APPROVED** [review]
LGTM, and tests are in

**Pull request merged**